llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -356,8 +385,8 @@ getPointerOperands(const Value &V, const DataLayout &DL,
     return {II.getArgOperand(0)};
   }
   case Instruction::IntToPtr: {
-    assert(isNoopPtrIntCastPair(&Op, DL, TTI));
-    auto *P2I = cast<Operator>(Op.getOperand(0));
+    Operator *P2I = getNoopOrExtPtrIntCastPair(&Op, DL, TTI);
+    assert(P2I != nullptr);
```

Comment:
Could you just have getNoopOrExtPtrIntCastPair end in unreachable if this cannot happen?

Reply:
I was a bit confused, `getNoopOrExtPtrIntCastPair` is also used to determine whether it is a `isAddressExpression` (line 335). Making it end in unreachable seems contradict with this purpose. Probably you mean separate `getNoopOrExtPtrIntCastPair` into two helper functions (also consider other two comments above), one tells it is `isNoopOrExtPtrIntCastPair` while the other one returns the desired operator?